File: src/com/sucy/go/ranks/impl/EffectivenessAlgorithm.java
package com.sucy.go.ranks.impl;

import com.sucy.go.Move;
import com.sucy.go.Pokemon;
import com.sucy.go.ranks.RankAlgorithm;

/**
 * PokemonGo © 2018
 * com.sucy.go.ranks.impl.EffectivenessAlgorithm
 */
public class EffectivenessAlgorithm extends RankAlgorithm {
    @Override
    public String getName() {
        return "effectiveness";
    }

    @Override
    public int compute(
            final Pokemon pokemon, final Move quickMove, final Move chargeMove) {
        final double dps = calcMoveDps(pokemon, quickMove, chargeMove);
        return (int)(dps * pokemon.attack() * pokemon.defense() * pokemon.stamina());
    }
}

File: site/pages/stats/EffectivenessAlgorithm.js
import Algorithm from './Algorithm'
import {POKEMON_LIST} from "app/data/Database";

class EffectivenessAlgorithm extends Algorithm {
  name = "effectiveness";

  compute(pokemon, quickMove, chargeMove, typeEffectiveness) {
    const dps = this.calcMoveDps(pokemon, quickMove, chargeMove, typeEffectiveness);
    return dps * pokemon.getAttack() * pokemon.getDefense() * pokemon.getStamina();
  }
}

const inst = new EffectivenessAlgorithm();
inst.rank(POKEMON_LIST);
export default inst
